Add source origin information to WKUIDelegate calls for badging

beidson opened the pull request:
#### a7347736572758d9f671498aaab7ec3b37ca88cb
<pre>
Add source origin information to WKUIDelegate calls for badging
<a href="https://bugs.webkit.org/show_bug.cgi?id=249618">https://bugs.webkit.org/show_bug.cgi?id=249618</a>
rdar://103536731

Reviewed by Chris Dumez.

API clients need to be able to decide whether to allow a badging update based on source origin.

In the case of the shared/service worker update coming through the _WKWebsiteDataStoreDelegate,
all the information was there.

Not adding it to the WKUIDelegate was an oversight.

This fixes that.

* Source/WebKit/PlatformMac.cmake:
* Source/WebKit/UIProcess/API/APIUIClient.h:
(API::UIClient::updateAppBadge):
(API::UIClient::updateClientBadge):
* Source/WebKit/UIProcess/API/Cocoa/WKSecurityOrigin.mm:
(-[WKSecurityOrigin isSameSiteAsOrigin:]):
(-[WKSecurityOrigin isSameSiteAsURL:]):
* Source/WebKit/UIProcess/API/Cocoa/WKSecurityOriginPrivate.h: Copied from Source/WebKit/UIProcess/API/Cocoa/WKSecurityOrigin.mm.
* Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.mm:
(WebKit::UIDelegate::setDelegate):
(WebKit::UIDelegate::UIClient::updateAppBadge):
(WebKit::UIDelegate::UIClient::updateClientBadge):
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::setAppBadge):
(WebKit::WebProcessProxy::setClientBadge):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Badging.mm:
(-[BadgeDelegate updatedAppBadge:fromOrigin:]):
(-[BadgeDelegate _webView:updatedAppBadge:fromSecurityOrigin:]):
(-[BadgeDelegate _webView:updatedClientBadge:fromSecurityOrigin:]):
(-[BadgeDelegate websiteDataStore:workerOrigin:updatedAppBadge:]):
(-[BadgeDelegate updatedAppBadge:]): Deleted.
(-[BadgeDelegate _webView:updatedAppBadge:]): Deleted.
(-[BadgeDelegate _webView:updatedClientBadge:]): Deleted.

Canonical link: <a href="https://commits.webkit.org/258123@main">https://commits.webkit.org/258123@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/61ece81639e41467c8b1ccf7731537e23d0c4792

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100966 "7 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10120 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34017 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110270 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/170532 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/104953 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/11051 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/991 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/93373 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108106 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106749 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/8359 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91615 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34965 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/90267 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/23014 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/77949 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/91446 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3795 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/24532 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/87543 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/1337 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3821 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/936 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/29249 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9932 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44035 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/90433 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5577 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5591 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/20234 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/Badging.mm:
```diff
@@ -462,4 +485,46 @@ - (void)websiteDataStore:(WKWebsiteDataStore *)dataStore workerOrigin:(WKSecurit
+{
+    auto configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
+
+    auto handler = [[TestURLSchemeHandler new] autorelease];
```
My understanding is that we're trying to avoid explicit autorelease like this and use RetainPtr instead, to make the transition to ARC easier. so I would `adoptNS()` here, just like you did above for the configuration.

cdumez reviewed (approved):
LGTM

beidson committed:
Add source origin information to WKUIDelegate calls for badging
https://bugs.webkit.org/show_bug.cgi?id=249618
rdar://103536731

Reviewed by Chris Dumez.

API clients need to be able to decide whether to allow a badging update based on source origin.

In the case of the shared/service worker update coming through the _WKWebsiteDataStoreDelegate,
all the information was there.

Not adding it to the WKUIDelegate was an oversight.

This fixes that.

* Source/WebKit/PlatformMac.cmake:
* Source/WebKit/UIProcess/API/APIUIClient.h:
(API::UIClient::updateAppBadge):
(API::UIClient::updateClientBadge):
* Source/WebKit/UIProcess/API/Cocoa/WKSecurityOrigin.mm:
(-[WKSecurityOrigin isSameSiteAsOrigin:]):
(-[WKSecurityOrigin isSameSiteAsURL:]):
* Source/WebKit/UIProcess/API/Cocoa/WKSecurityOriginPrivate.h: Copied from Source/WebKit/UIProcess/API/Cocoa/WKSecurityOrigin.mm.
* Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.mm:
(WebKit::UIDelegate::setDelegate):
(WebKit::UIDelegate::UIClient::updateAppBadge):
(WebKit::UIDelegate::UIClient::updateClientBadge):
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::setAppBadge):
(WebKit::WebProcessProxy::setClientBadge):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Badging.mm:
(-[BadgeDelegate updatedAppBadge:fromOrigin:]):
(-[BadgeDelegate _webView:updatedAppBadge:fromSecurityOrigin:]):
(-[BadgeDelegate _webView:updatedClientBadge:fromSecurityOrigin:]):
(-[BadgeDelegate websiteDataStore:workerOrigin:updatedAppBadge:]):
(-[BadgeDelegate updatedAppBadge:]): Deleted.
(-[BadgeDelegate _webView:updatedAppBadge:]): Deleted.
(-[BadgeDelegate _webView:updatedClientBadge:]): Deleted.

Canonical link: https://commits.webkit.org/258123@main
